classes/services/client.php

```diff
@@ -147,13 +157,20 @@ protected function request( $method, $endpoint, $args = [] ) {
 			$body = true;
 		}
 
-		// Return with no content on successfull deletion of domain from service.
+		// Return with no content on successful deletion of domain from service.
 		if ( 204 === $response_code ) {
 			$body = true;
 			return $body;
 		}
 
-		$body = json_decode( $body );
+		/**
+		 * product/widget endpoint returns javascript hence we
+		 * escape decoding for the data received in body from
+		 * this endpoint.
+		 */
+		if ( ! strpos( $endpoint, 'widget' ) ) {
+			$body = json_decode( $body );
+		}
```

Comment:
why do we need this? we will never call the widget from the service client directly